### PR TITLE
Refactor sink.rb

### PR DESCRIPTION
**Pull request by artem-shelkovnikov**
Things done in this PR:

- Base class for sinks
- Each sink implementation has its own file
- Moved sinks to `core/output_sink` namespace

### CODE REVIEW

File: lib/core/output_sink/console_sink.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'core/output_sink'
+require 'utility/logger'
+
+module Core::OutputSink
+  class ElasticSink < Core::OutputSink::BaseSink
```
Comment (wangch079) [inline lib/core/output_sink/console_sink.rb:13]:
```suggestion
  class ConsoleSink < Core::OutputSink::BaseSink
```

File: lib/core/output_sink/console_sink.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    def ingest(_document)
+      print_header 'Got a single document:'
+    end
```
Comment (wangch079) [inline lib/core/output_sink/console_sink.rb:16]:
```suggestion
    def ingest(document)
      print_header 'Got a single document:'
      Utility::Logger.info document
    end
```

To be consistent with `ingest_multiple(documents)

**Review by wangch079 — APPROVED** [review]
LGTM except for a typo.

Agree with what @seanstory suggested. `Elastic` is the company name, we should rename it. We have `EsClient`, so I think either we rename both to `ElasticsearchClient` and `ElasticsearchSink` or we use `EsClient` or `EsSink`, to be consistent.

File: lib/core/output_sink/console_sink.rb (hunk 2 above)
Comment (artem-shelkovnikov) [reply lib/core/output_sink/console_sink.rb]: Good call, fixed!

**Comment by artem-shelkovnikov** [comment]
> Do these have specs?

Not yet, planning to add basic tests to everything as the last set of tasks for connector, now focussing on functionality and ability to throw away/rewrite major pieces of code quickly.

I'm a big fan of tests, so I'll close as much as possible in the last week before release :)

**Comment by artem-shelkovnikov** [comment]
> Agree with what @seanstory suggested. Elastic is the company name, we should rename it. We have EsClient, so I think either we rename both to ElasticsearchClient and ElasticsearchSink or we use EsClient or EsSink, to be consistent.

Agreed, renamed to EsSink to be consistent with EsClient. I like `ElasticsearchX` more, but this can be renamed later.

**Pull request merged**